docs(reference/hooks): add information about prehandler (#5163)
fastify/fastify: docs/Reference/Hooks.md

```

### preHandler

The `preHandler` hook allows you to specify a function that is executed before

> routes' handler.

Reviewer: ```suggestion
a routes's handler.
```

```

### preHandler

> The `preHandler` hook allows you to specify a function that is executed before routes' handler.

Reviewer 2: ```suggestion
The `preHandler` hook allows you to specify a function that is executed before
routes' handler.
```

> The `beforeHandler` option allows you to specify a function that is executed before the main route handler.

Reviewer 3: ```suggestion
The `preHandler` hook allows you to specify a function that is executed before routes' handler.
```
